src/other/tables.jl

```diff
@@ -0,0 +1,39 @@
+using Tables, IteratorInterfaceExtensions
+
+Tables.istable(::Type{<:AbstractDataFrame}) = true
+Tables.columnaccess(::Type{<:AbstractDataFrame}) = true
+Tables.columns(df::AbstractDataFrame) = df
+Tables.rowaccess(::Type{DataFrame}) = true
+Tables.rows(df::DataFrame) = eachrow(df)
+
+Tables.schema(df::AbstractDataFrame) = Tables.Schema(names(df), eltypes(df))
+Tables.schema(df::DFRowIterator) = Tables.schema(df.df)
+
+getvector(x::AbstractVector) = x
+getvector(x) = collect(x)
+fromcolumns(x) = DataFrame(Any[getvector(c) for c in Tables.eachcolumn(x)], collect(propertynames(x)))
+
+function DataFrame(x::T) where {T}
+    if Tables.istable(T)
+        return fromcolumns(Tables.columns(x))
+    end
+    y = IteratorInterfaceExtensions.getiterator(x)
+    yT = typeof(y)
+    if Base.isiterable(yT)
```

Comment:
It'd be ideal to avoid the dependency on TableTraits.jl. I can't see any difference between what's here and what's in TableTraits?

Reply:
Lots of sources in the Queryverse add a method to ``TableTraits.isiterabletable``, and you won't catch that with your own implementation.

Would it be so bad to take a dependency on TableTraits.jl? The package is specifically designed to not cause any problems, the whole code is https://github.com/queryverse/TableTraits.jl/blob/master/src/TableTraits.jl. I'm also making a guarantee that a) I will never ever make a breaking release for that package (unless julia 2.0.0 forces me to) and b) I won't take any further dependencies on. I can (maybe, a little bit) understand a reluctance to take a dependency on DataValues.jl, but TableTraits.jl I think is harmless. The package also has a long track record of not breaking anything.

I'm also realizing that the call to ``IteratorInterfaceExtensions.getiterator(x)`` is probably not what you want here. In general, the sequence ``y = IteratorInterfaceExtensions.getiterator(x); yT = typeof(y); Base.isiterable(yT)`` should be written as ``IteratorInterfaceExtensions.isiterable(x)`` for a much more efficient story. But even that is not really good, because really the way to properly hook into the Querverse there is to use ``TableTraits.isiterabletable``.

Reply:
Ok, I'm not quite sure I understand why doing `TableTraits.isiterabletable` explicitly is really better than what we had (which was doing the exact same fallback checks); are there queryverse sources that overload `isiterabletable` but wouldn't satisfy the fallback definition? Doesn't that seem troubling from a "clean interface" perspective? Anyway, not really the point here.

I've added the TableTraits.jl dependency; I agree that it's a very small dependency and simplifies things a bit w/ the queryverse integration. That said, the feeling among DataFrames admins is that DataFrames should really only need to support a single table interface long-term. The hope is that Tables.jl support continues to grow and given that it would be about 4 lines of code for any queryverse source to implement the interface, it would be great to be able to standardize.

So with this PR, DataFrames will have dependencies on DataStreams, Tables, and TableTraits, but the long-term goal is that only Tables.jl will be needed.